### PR TITLE
feat(custom-metric): apply create api & add unit field

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci`, minor refactoring, etc.)
- [ ] Need discussion
- [x] Not that difficult
- [ ] Approved feature branch merge to master

### Description
SSIA

image

### CODE REVIEW

File: apps/web/src/services/asset-inventory/components/AddCustomMetricModal.vue [hunk 1]
```diff
@@ -18,24 +30,37 @@ const props = withDefaults(defineProps<Props>(), {
 });
 
 const emit = defineEmits<{(e: 'update:visible', value: boolean): void}>();
+const metricExplorerPageStore = useMetricExplorerPageStore();
+
+const storeState = reactive({
+    metricList: computed(() => metricExplorerPageStore.state.metricList),
```
Comment (yuda110) [inline apps/web/src/services/asset-inventory/components/AddCustomMetricModal.vue:36]: suggestion: Why don't you change this to `metricNameList(string[])`?

```typescript
metricNameList: computed(() => metricExplorerPageStore.state.metricList.map((metric) => metric.name)),
```

Comment (piggggggggy) [reply apps/web/src/services/asset-inventory/components/AddCustomMetricModal.vue:36]: Good idea!

File: apps/web/src/services/asset-inventory/components/AddCustomMetricModal.vue [hunk 2]
```diff
@@ -48,38 +73,95 @@ const {
...
+        ErrorHandler.handleError(e);
+    }
+};
+
+/* Method */
+const resetForm = () => {
+    setForm('name', '');
+    setForm('unit', '');
+    setForm('code', '');
+};
```
Comment (yuda110) [inline apps/web/src/services/asset-inventory/components/AddCustomMetricModal.vue:106]: nitpick: there's `initForm()` in `useFormValidator` hook

Comment (piggggggggy) [reply apps/web/src/services/asset-inventory/components/AddCustomMetricModal.vue:105]: Oh thanks~!

**Review by yuda110 — APPROVED** [review]
good good

**Pull request merged**